src/Data/Generic.purs

```diff
@@ -277,6 +282,7 @@ data GenericSignature
   | SigChar
   | SigArray (Unit -> GenericSignature)
   | SigUnit
+  | SigVoid
```

Comment:
We can do this without a breaking change if we use `SigProd "Data.Void.Void" []` here.